### PR TITLE
feat: move tree to root

### PR DESCRIPTION
**Pull request by caarlos0**
closes #339 
closes #341

### CODE REVIEW

**Review by bashbunni — APPROVED** [review]

**Review by aymanbagabas — APPROVED** [review]

**Comment by meowgorithm** [comment]
If you don't mind, going to look at a few last things today before we merge this one.

**Comment by bashbunni** [comment]
Hey @meowgorithm any changes on this one?

**Comment by meowgorithm** [comment]
Let’s get the tree examples back in as well — I'm looking at stuff now.

**Comment by bashbunni** [comment]
@meowgorithm kk no rush. Just added the examples back

**Review by meowgorithm — APPROVED** [review]
Feeling good! Let's get this done ⚡️

**Comment by bashbunni** [comment]
Removed `JoinVertical` from the final rendered element as it was writing unstyled whitespace to the final string. It does that by design to appropriately align elements in the block, but because it doesn't have access to the style, the whitespace can't be rendered. That said, I haven't seen this issue outside of the tree examples, so it's not pressing. 

If we did want to fix this we could replace this whitespace rendering with calling `Align` on the style which can render the whitespace properly or fix this with our compositor in a future version of lipgloss

**Review by bashbunni — CHANGES_REQUESTED** [review]
TODO before merge
- [ ] tidy godoc (some tree docs mention list)
- [x] show an example of unique styling of the root element (this might be broken)
- [ ] add tree info to `README.md`

**Comment by bashbunni** [comment]
I thought the lipgloss root styling might be broken because if we do 

```go
ItemStyleFunc(func(children tree.Children, i int) lipgloss.Style {
    if children.At(i).Value() == "Nyx" {                          
        return itemStyle.Background(lipgloss.Color("#04B575"))    
    }                                                             
    return itemStyle                                              
})                                                                
```
we get 
```
%!v(PANIC=String method: runtime error: invalid memory address or nil pointer dereference)
``` 
because when we're on the root item, it's `children.At(-1)` which returns `nil`, then we attempt to call `Value()` on it. 

_As-is we will always need to do a root value check in `ItemStyleFunc` to avoid this..._

**Comment by bashbunni** [comment]
We might even want to see about separating out `RootStyle` from `ItemStyle` 💭

**Comment by caarlos0** [comment]
> We might even want to see about separating out `RootStyle` from `ItemStyle` 💭

i think that makes more sense yes

File: README.md [hunk 1]
```diff
@@ -562,6 +562,104 @@ for i := 0; i < repeat; i++ {
...
+// .
+// ├── A
+// ├── B
+// └── C
+```
+
+Trees have the ability to nest.
+
+```go
+t := tree.New().
```
Comment (bashbunni) [inline README.md:595]: I can use the shorthand `tree.Root(".")...` and get rid of `New()` (not needed)

**Review by bashbunni — APPROVED** [review]

**Review by aymanbagabas — APPROVED** [review]

**Review by meowgorithm — APPROVED** [review]

**Pull request merged**